src/aiida/orm/nodes/data/structure.py

```diff
@@ -1762,21 +1762,25 @@ def _get_object_pymatgen_structure(self, **kwargs):
         :return: a pymatgen Structure object corresponding to this
           :py:class:`StructureData <aiida.orm.nodes.data.structure.StructureData>`
           object
-        :raise ValueError: if periodic boundary conditions does not hold
-          in at least one dimension of real space; if there are partial occupancies
-          together with spins (defined by kind names ending with '1' or '2').
+        :raise ValueError: if the cell is not set (i.e. is the default one);
+          if there are partial occupancies together with spins
+          (defined by kind names ending with '1' or '2').
 
         .. note:: Requires the pymatgen module (version >= 3.0.13, usage
             of earlier versions may cause errors)
         """
+        import numpy as np
+        from pymatgen.core.lattice import Lattice
         from pymatgen.core.structure import Structure
 
-        if self.pbc != (True, True, True):
```

Comment:
I believe you still need this check even if you check for default cell, because this function should raise when you pass it a non-periodic structure with `self.pbc == (False, False, False)`

Reply:
I do not agree, since pymatgen Structure class can handle _any_ periodicity, provided you set a proper cell (e.g. a molecule in a box). Then the `get_pymatgen` function is dedicated to handle this, so that if it is 0D then you will get a pymatgen Molecule class instance, but if you ask to get a pymatgen Structure, you should get so (if the cell is not singular).

Reply:
I see, that makes sense. But in that case you might need to modify a couple of docstrings in this file. Specifically for `get_pymatgen_structure` (maybe others?).